cpp/src/arrow/compute/exec.cc

```diff
@@ -1022,25 +1048,7 @@ Status CheckCanExecuteChunked(const VectorKernel* kernel) {
 class VectorExecutor : public KernelExecutorImpl<VectorKernel> {
  public:
   Status Execute(const ExecBatch& batch, ExecListener* listener) override {
-    // Some vector kernels have a separate code path for handling
-    // chunked arrays (VectorKernel::exec_chunked) so we check if we
-    // have any chunked arrays. If we do and an exec_chunked function
-    // is defined then we call that.
-    bool have_chunked_arrays = false;
-    for (const Datum& arg : batch.values) {
-      if (arg.is_chunked_array()) have_chunked_arrays = true;
-    }
-
-    output_num_buffers_ = static_cast<int>(output_type_.type->layout().buffers.size());
-
-    // Decide if we need to preallocate memory for this kernel
-    validity_preallocated_ =
```

Comment:
We could use more precise pre-allocation strategy by check the input batches like what `ScalarExecutor` do.

Reply:
FWIW I'm working on Take at the moment and setting the chunked exec kernels https://github.com/apache/arrow/pull/41700

Reply:
Thanks for the reminder, I will take a look.
Current PR just optimize the execution logic of chunked-array from the NULL pre-allocation part.